tools/cabana/streams/routes.cc

```diff
@@ -92,40 +102,83 @@ void RoutesDialog::fetchRoutes() {
   if (device_list_->currentIndex() == -1 || device_list_->currentData().isNull())
     return;
 
-  route_list_->clear();
-  route_list_->setEmptyText(tr("Loading..."));
+  currentRoutesList()->clear();
+  currentRoutesList()->setEmptyText(tr("Loading..."));
 
   // Construct URL with selected device and date range
   auto dongle_id = device_list_->currentData().toString();
-  QDateTime current = QDateTime::currentDateTime();
-  QString url = QString("%1/v1/devices/%2/routes_segments?start=%3&end=%4")
-                    .arg(CommaApi::BASE_URL).arg(dongle_id)
-                    .arg(current.addDays(-(period_selector_->currentData().toInt())).toMSecsSinceEpoch())
-                    .arg(current.toMSecsSinceEpoch());
+  QString url = QString("%1/v1/devices/%2")
+                      .arg(CommaApi::BASE_URL)
+                      .arg(dongle_id);
+
+  route_requester_ = new OneShotHttpRequest(this);
```

Comment:
had to recreate the http object here since the `reply` is now being set to nullptr, this will fail if I try to reuse it.

Reply:
this was crashing the app.